### ISSUE TITLE
CS1591 & CS1711 not respecting inheritdoc+cref

### ISSUE THREAD

**Issue by ZacharyPatten** [dotnet/roslyn#42392]
## Topic

`inheritdoc` supports `cref` so that you can inherit the documentation of specific members, but `CS1591` & `CS1711` are triggering when you have them on methods designed to be used with `inheritdoc` in the same assembly.

## Possible Resolution

It might be a good idea to check for `inheritdoc` in the same assembly before firing `CS1591` & `CS1711`.

## Example

```cs
using System;
class Class
{
#pragma warning disable CS1711 // XML comment has a typeparam tag, but there is no type parameter by that name
#pragma warning disable CS1572 // XML comment has a param tag, but there is no parameter by that name
	/// <summary>Documentation.</summary>
	/// <typeparam name="A">Documentation.</typeparam>
	/// <param name="a">Documentation.</param>
	/// <param name="b">Documentation.</param>
	/// <param name="c">Documentation.</param>
	/// <returns>Documentation.</returns>
	[Obsolete("This method is for documentation only.", true)]
	internal static void Method_XML() => throw new Exception("This method is for documentation only.");
#pragma warning restore CS1572 // XML comment has a param tag, but there is no parameter by that name
#pragma warning restore CS1711 // XML comment has a typeparam tag, but there is no type parameter by that name
	/// <inheritdoc cref="Method_XML"/>
	public static void Method() => throw new NotImplementedException();
	/// <inheritdoc cref="Method_XML"/>
	public static void Method(int a) => throw new NotImplementedException();
	/// <inheritdoc cref="Method_XML"/>
	public static void Method<A>() => throw new NotImplementedException();
	/// <inheritdoc cref="Method_XML"/>
	public static void Method<A>(A a) => throw new NotImplementedException();
	/// <inheritdoc cref="Method_XML"/>
	public static void Method(int a, int b) => throw new NotImplementedException();
	/// <inheritdoc cref="Method_XML"/>
	public static int Method(int a, float c) => throw new NotImplementedException();
}
```

## Example 2

Here is some of my code where I am using this pattern if it helps:
https://github.com/ZacharyPatten/Towel/blob/ab53ccda326b953432eb6a1881115a7d9d518c73/Sources/Towel/Sort.cs#L11

**Comment by ZacharyPatten** [comment]
@sharwell

**Comment by sharwell** [comment]
Moving to @dotnet/roslyn-compiler. We don't have control over these warnings in the IDE layers.

**Comment by ZacharyPatten** [comment]
### Related to https://github.com/dotnet/roslyn/issues/42392

- https://github.com/dotnet/roslyn/issues/42392 false positives on the parent of the `inheritdoc`
- https://github.com/dotnet/roslyn/issues/40325 +`CS1712` false positives on the child of the `inheritdoc`

> Sorry for the duplicate comment, but I wanted don't want people to miss this if they only view one or the other.

**Comment by ZacharyPatten** [comment]
> ```
> #pragma warning disable CS1711 // XML comment has a typeparam tag, but there is no type parameter by that name
> #pragma warning disable CS1572 // XML comment has a param tag, but there is no parameter by that name
> #pragma warning disable CS1734 // XML comment has a paramref tag, but there is no parameter by that name
> #pragma warning disable CS1735 // XML comment has a typeparamref tag, but there is no type parameter by that name
> ```
> 
> I would not expect any of these diagnostics to be considered by inheritdoc. If you want to use one primary method to document other overloads, the documentation should be placed on the overload with the most parameters, and should only include the documentation for the actual elements that appear in that signature.

Unfortunately that just isn't the ideal solution. If you do that you will end up with a spider web of `inheritdoc` that is riddled with duplicates and hard to maintain. Having a single source of documentation for all similar methods is just better.

**Comment by sharwell** [comment]
Since you're using a documentation pattern that deviates quite substantially from the original language rules, it probably makes sense to either 1) disable the compiler diagnostics and write your own set of analyzers that cover the functionality or 2) write a diagnostic suppressor that understands the coding patterns you are using and suppresses the compiler diagnostics at the locations where you would consider them false positives.
